uaclient/exceptions.py

```diff
@@ -70,6 +72,9 @@ class LockHeldError(UserFacingError):
     """
 
     def __init__(self, lock_request: str, lock_holder: str, pid: int):
+        self.lock_request = lock_request
+        self.lock_holder = lock_holder
+        self.lock_holder_pid = pid
```

Comment:
Are we using those variables somewhere else ?

Reply:
Only in the test :/

If you think this is unnecessary I'm open to removing it

Reply:
Maybe we could directly check the `msg` variable of the exception instead. It will be a proxy for those variables

Reply:
Yeah that's fair, I'll make that change

Reply:
Oops actually I also used these in `SpinLock.__enter__` for the log message there, so I think we should keep these

Reply:
One thing we can do is rely on the message of the exception and just append the `spinning...` part to it.
But maybe this is unnecessary. Let me know what you think

Reply:
Sure we can do that

Reply:
Updated!